File: python/EyeScanner.py
import Distance
import Stepper

# front eye controller
class EyeScanner:

    # util method to get stepper motor to front direction
    def calibrate(self):
        i     = 0 # security to avoid complete rotation
        check = self.dist.measure()
        print(str(check))
        while (i < 200 and check > 8.0 and check < 1000.0):
            self.stepper.rotate(5,-1)
            check = self.dist.measure()
        self.stepper.rotate(self.caldist,1)
        self.pos = 0
        return i < 200

    # first method to scan minimum dist on given step area (scan from (0 - mstep) to (msteps)) 0 is from calibration
    # one step is 5 steps on stepper motor
    def simple_scan(self,msteps):
        minval = 2000.0
        self.stepper.rotate(msteps*10,1)
        for i in range(0-msteps,msteps):
            val = self.dist.measure()
            if (val < minval):
                minval = val
                print(str(minval))
            self.stepper.rotate(10,-1)
        self.stepper.rotate(msteps*10,1)
        return minval


    def __init__(self,calibratevalue=405):
        self.caldist = calibratevalue
        self.pos     = 0
        self.stepper = Stepper.Stepper()
        self.dist    = Distance.Distance()

File: cpp/hardware/Eyescanner.cpp
#include <chrono>
#include <wiringPi.h>
#include "Eyescanner.h"

//
// default constructor
//
EyeScanner::EyeScanner() {
}

// calibration
bool EyeScanner::calibration() {
  int i=0;
  double check = this->dist.measure();
  while (i < 200 && check > 8.0) {
    this->stepper.rotate(2,-1);
    check = this->dist.measure();
  }
  this->stepper.rotate(this->calvalue,1);
  position = 0;
  return i < 200;
}

// simple min dist scan for given steps
double EyeScanner::simple_scan(int steps) {
  double minval = 2000.0;
  double val;

  this->stepper.rotate(steps*10,1);
  for (int i=(0-steps);i<steps;i++) {
    val = this->dist.measure();
    if (val < minval) {
      minval = val;
    }
    this->stepper.rotate(10,-1);
  }
  this->stepper.rotate(steps*10,1);
  return minval;
}
